### PR TITLE
[refman] Misc fixes (indentation, whitespace, notation syntax)

### PR DESCRIPTION
**Pull request by cpitclaudel**

### CODE REVIEW

**Comment by cpitclaudel** [comment]
Btw, I need help with the following:

```
Add Parametric Morphism (x1 : T1) ... (xk : Tk) : (f t1 ... tn) with signature sig as @ident
Add Parametric Relation (x1 : T1) ... (xn : Tk) : (A t1 ... tn) (Aeq t′1 ... t′m) {? reflexivity proved by refl} {? symmetry proved by sym} {? transitivity proved by trans} as @ident
Derive Dependent Inversion @ident with forall (x:T), I t Sort @sort
Derive Dependent Inversion_clear @ident with forall(x:T), I t Sort @sort
Derive Inversion @ident with forall (x : T), I t Sort sort
Derive Inversion_clear @ident with forall (x:T), I t Sort @sort
Instance @ident {? @binders} : @class t1 … tn {? | priority } := { field1 := b1 ; …; fieldi := bi }
Instance @ident {? @binders} : forall {? @binders}, @class @term__1 … @term__n {? | priority } := @term
```

One problem here is that we don't have a way to name an `@ident` for easier reference in the text.  For example this:

```rst
.. cmdv:: Record {? >} @ident {? @binders} : @sort := {? @ident} { {+; @ident :{? >} @term } }

   The first identifier :token:`ident` is the name of the defined record and
   :token:`sort` is its type. The optional identifier after ``:=`` is the name
   of the constructor (it will be :n:`Build_@ident` if not given).
   The other identifiers are the names of the fields, and :token:`term`
   are their respective types.
```

I wonder if it would read better like this:

```rst
.. cmdv:: Record {? >} @{name:ident} {? @binders} : @sort := {? @{constructor:ident} { {+; @{fieldname:ident} :{? >} @{fieldtype:term} } }

   The identifier ``name`` is the name of the defined record and
   :token:`sort` is its type. ``constructor`` is the name
   of the constructor (it will be :n:`Build_@{name}` if not given).
   The ``fieldname``\ s are the names of the fields, and ``fieldtype``\ s
   are their respective types.
```

I'm not fully convinced. The complexity might not be worth it.

**Comment by cpitclaudel** [comment]
> I'm not fully convinced. The complexity might not be worth it.

An alternative could be to define aliases locally; something like `.. ident:: @Aeq = @ident`.

**Comment by cpitclaudel** [comment]
Updated with two more fixes

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 1]
```diff
@@ -170,12 +170,12 @@ compatibility constraints.
 Adding new relations and morphisms
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. cmd::  Add Parametric Relation (x1 : T1) ... (xn : Tk) : (A t1 ... tn) (Aeq t′1 ... t′m) {? reflexivity proved by refl} {? symmetry proved by sym} {? transitivity proved by trans} as @ident
+.. cmd::  Add Parametric Relation @binders : (A t1 ... tn) (Aeq t′1 ... t′m) {? reflexivity proved by @ident} {? symmetry proved by @ident} {? transitivity proved by @ident} as @ident
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: The proofs can actually be terms.

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 2]
```diff
@@ -219,15 +219,16 @@ replace terms with related ones only in contexts that are syntactic
 compositions of parametric morphism instances declared with the
 following command.
 
-.. cmd:: Add Parametric Morphism (x1 : T1) ... (xk : Tk) : (f t1 ... tn) with signature sig as @ident
+.. cmd:: Add Parametric Morphism @binders : (@ident {+ @term}) with signature @sig as @ident
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: Is `sig` a defined token?

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 3]
```diff
@@ -219,15 +219,16 @@ replace terms with related ones only in contexts that are syntactic
...
 
-   This command declares ``f`` as a parametric morphism of signature ``sig``. The
-   identifier :token:`ident` gives a unique name to the morphism and it is used as
-   the base name of the typeclass instance definition and as the name of
-   the lemma that proves the well-definedness of the morphism. The
-   parameters of the morphism as well as the signature may refer to the
-   context of variables. The command asks the user to prove interactively
-   that ``f`` respects the relations identified from the signature.
+   This command declares a parametric morphism :n:`@ident {+ @term}` of
+   signature :ident:`sig`.  The final identifier :token:`ident` gives a unique
```
Comment (Zimmi48) [inline doc/sphinx/addendum/generalized-rewriting.rst]: What is `:ident:`? Is it an inline definition of token `sig` similar to `:production:`?

**Review by Zimmi48 — COMMENTED** [review]
What are the whitespace changes?

> An alternative could be to define aliases locally; something like `.. ident:: @Aeq = @ident`.

In some cases, this could turn out to be useful. In most cases though, it is sufficient to just add indices to the various repeated tokens `@ident__1`, `@ident__2` for easier reference in the text later on.

File: doc/sphinx/addendum/generalized-rewriting.rst (hunk 1 above)
Comment (cpitclaudel) [reply doc/sphinx/addendum/generalized-rewriting.rst]: Thanks

File: doc/sphinx/addendum/generalized-rewriting.rst (hunk 3 above)
Comment (cpitclaudel) [reply doc/sphinx/addendum/generalized-rewriting.rst]: No, it's me messing up.

**Comment by cpitclaudel** [comment]
The whitespace changes are due to the original author using SIX-PER-EM SPACE (U+2006).  Not sure why they did that.

File: doc/sphinx/addendum/generalized-rewriting.rst (hunk 2 above)
Comment (cpitclaudel) [reply doc/sphinx/addendum/generalized-rewriting.rst]: Apparently not :'(

**Comment by cpitclaudel** [comment]
All fixed, I think

**Comment by cpitclaudel** [comment]
Thanks for your careful proofreading!

**Review by Zimmi48 — APPROVED** [review]
I know nothing about `Derive Inversion` so wouldn't mind someone checking this part before I merge.

**Comment by cpitclaudel** [comment]
I added four more small changes, but let me know if you'd prefer these to be a separate PR.  I made them separate commits to make the review easier

File: doc/sphinx/language/gallina-extensions.rst [hunk 4]
```diff
@@ -85,7 +85,7 @@ To build an object of type :token:`ident`, one should provide the constructor
 
   .. productionlist::
     record_term : {| [`field_def` ; … ; `field_def`] |}
-    field_def : name [binders] := `record_term`
+    field_def : `name` [`binders`] := `record_term`
```
Comment (Zimmi48) [inline doc/sphinx/language/gallina-extensions.rst]: I am not aware of `name` being a defined token, so it should probably be `ident` instead.
As for the left hand-side of the `:=` it should probably be `term` and not `record_term`.

File: doc/sphinx/user-extensions/syntax-extensions.rst [hunk 5]
```diff
@@ -327,22 +327,29 @@ symbols.
 Reserving notations
 ~~~~~~~~~~~~~~~~~~~
 
-A given notation may be used in different contexts. Coq expects all
-uses of the notation to be defined at the same precedence and with the
-same associativity. To avoid giving the precedence and associativity
-every time, it is possible to declare a parsing rule in advance
-without giving its interpretation. Here is an example from the initial
-state of Coq.
+.. cmd:: Reserved Notation @string {*, @modifiers}
```
Comment (Zimmi48) [inline doc/sphinx/user-extensions/syntax-extensions.rst]: For consistency with #10118, I suggest:

```suggestion
.. cmd:: Reserved Notation @string {? (@modifiers) }
```

File: doc/sphinx/language/gallina-extensions.rst (hunk 4 above)
Comment (cpitclaudel) [reply doc/sphinx/language/gallina-extensions.rst]: `name` is `ident` or `_`, so in this case `ident` is indeed right; thanks! You're also right about `record_term`.

**Comment by cpitclaudel** [comment]
Rebased.

**Pull request merged**